src/zero_copy/error.rs

```diff
@@ -166,29 +326,27 @@ where
     I::Token: fmt::Debug,
 {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "found ")?;
-        write_token_debug(f, &self.found)?;
-        write!(f, " at {:?}, expected ", self.span)?;
-        match &self.expected[..] {
-            [] => write!(f, "something else")?,
-            [expected] => write_token_debug(f, expected)?,
-            _ => {
-                write!(f, "one of ")?;
-                for expected in &self.expected[..self.expected.len() - 1] {
-                    write_token_debug(f, expected)?;
-                    write!(f, ", ")?;
-                }
-                write!(f, "or ")?;
-                write_token_debug(f, self.expected.last().unwrap())?;
-            }
-        }
-        Ok(())
+        self.inner_fmt(f, I::Token::fmt, I::Span::fmt)
+    }
+}
+
+impl<I: Input + ?Sized> fmt::Display for Rich<I>
+where
+    I::Span: fmt::Display,
```

Comment:
This requirement is a little restrictive, since most spans are `Range` - opinions on whether I should change it to debug, or do something else?

Reply:
I think this is most correct, even if a little annoying. Perhaps we should provide our own `SimpleSpan` type? That way we can also add a `Copy` bound to it, as well as a `Display` impl.

Reply:
That sounds like a good idea. It would need to entirely replace `Range` - since `Span` is associated with `Input`, but I'm not entirely against that, especially as it's not too hard to make it pretty trivially convertable, and `zero_copy` breaks the world anyways.